### ISSUE TITLE
iPadOS Connect/Auth fails since 2024.913 - received close message during auth phase

### ISSUE THREAD

**Issue by rct** [home-assistant/iOS#3163]
Model Name: iPad Pro (10.5 inch) (iPad7,4) 
Software Version: 17.7.1
App version: 

**Home Assistant Core Version**
2024.11.1

(Also happened with 2024.10.x)

**Describe the bug**
iPad with TestFlight, 2024.9.4 (2024.905) and earlier connect/auth fine, all 2024.10 versions and 2024.11 (2024.951) fail to connect to any of my 3 Home Assistant Instances.   If I fall back to 2024.9.4, it connects successfully to all 3. 

It seems to go into a connect/auth/close loop generating:

`(Home Assistant/2024.11 (io.robbie.HomeAssistant; build:2024.951; iPadOS 17.7.1)): Disconnected: Received close message during auth phase`

On one instance it generates a login failed persistent notification. but that doesn't show up on the other instances. 

**To Reproduce**

**Expected behavior**

**Screenshots**

**Additional context**

Log from 2024.11. Hass instance

```
2024-11-15 09:44:41.352 WARNING (MainThread) [homeassistant.components.websocket_api.http.connection] [140583947258608] from 192.168.x.y (Mozilla/5.0 (iPad; CPU OS 17_7_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Home Assistant/2024.11 (io.robbie.HomeAssistant; build:2024.951; iPadOS 17.7.1) Mobile/HomeAssistant, like Safari): Disconnected: Received close message during auth phase
2024-11-15 09:44:41.384 WARNING (MainThread) [homeassistant.components.websocket_api.http.connection] [140583947258608] from 192.168.x.y (Home Assistant/2024.11 (io.robbie.HomeAssistant; build:2024.951; iPadOS 17.7.1)): Disconnected: Received close message during auth phase
2024-11-15 09:44:41.438 WARNING (MainThread) [homeassistant.components.websocket_api.http.connection] [140583217311472] from 192.168.x.y (Home Assistant/2024.11 (io.robbie.HomeAssistant; build:2024.951; iPadOS 17.7.1)): Disconnected: Received close message during auth phase
2024-11-15 09:44:41.463 WARNING (MainThread) [homeassistant.components.websocket_api.http.connection] [140583217307440] from 192.168.x.y(Home Assistant/2024.11 (io.robbie.HomeAssistant; build:2024.951; iPadOS 17.7.1)): Disconnected: Received close message during auth phase
```

## Edit/Update 2024-11-27

Problem started with 2024.10 build 2024.927.   The first two 2024.10 builds 2024.911 and 2024.913 did NOT have the problem.

The beta 2024.11 - build 2024.978 seems to fix the problem.

**Comment by bgoncal** [comment]
Are you using the internal or external IP?
If internal, have you defined your local SSID (network name) inside the internal URL screen config? 

There is a change in the app that it will only use internal IP if the network name if defined.

**Comment by rct** [comment]
My server configuration details in the app
* For each of the 3 defined servers both Internal and External URLs are defined.  For 2 of the servers, the external URL is Nabu Casa cloud.  For the one that doesn't have a subscription the internal and external URLs are the same. 
* I have SSIDs defined under each server's internal URL.
* I am on my internal network with one of those SSIDs after upgrading when it fails to connect.  As soon as I downgrade to 2024.9 it connects again.
* Local push is turned off.  I turned it off a year or so ago because of battery drain.

Note: **It gets stuck in a loop trying to connect**  It logged `Disconnected: Received close message during auth phase` over 300 times during the course of a minute or two.

**Comment by bgoncal** [comment]
Thanks, If you check which URL it is connected to, does it says "internal"? (Tell me per server please)

**Comment by rct** [comment]
Hmm, I went to try this again today with the most recent 2024.11 build. It connected just fine. I haven't changed my configuration on either the iPad or the server. 

With regards to checking which URL it is connected to, I could be wrong, but I believe it said local, but the WebSocket line was rapidly flashing between the different connecting, authorizing, disconnected states.

**Comment by bgoncal** [comment]
I am closing this issue but it would be interesting to debug it again if it keeps happening.

**Comment by rct** [comment]
@homeassistant reopen

**Comment by rct** [comment]
So I thought things were fine with 2024.11, since it connected after updating. However, I tried to reopen the app later and got back in the same state where it wouldn't connect and was back on the "Retrying in 59 seconds...." screen.

I went into the server settings, verified it was the internal URL it was trying to connect to, saw the flashing "Conncting/Authorizing" messages next to WebSocket. 

Over the course of a minute, Home Assistant logged `(Home Assistant/2024.11 (io.robbie.HomeAssistant; build:2024.951; iPadOS 17.7.1)): Disconnected: Received close message during auth phase` 136 times.

**Comment by bgoncal** [comment]
All of your servers have this issue or just one?

**Comment by rct** [comment]
Once I've upgraded and get the connection problem, I've tried switching between all three servers and can't get a connection.

**Comment by rct** [comment]
Please let me know what I can be doing to collect useful information.

Also note that the "state"  displayed in the app, the "Unable to connect, retrying in 60 seconds" does NOT reflect what the app is actually doing:

* It appears to be still connecting several times a second.
* When I try to switch servers it still keeps trying to connect the previous server
* If I do manage to get successfully connected to a server, it still appears to be trying to connect to the other two servers in the background (observed by tailing the app logs and seeing 'the Received close message during auth phase' repeating.

One other odd thing I've noticed, on the "Unable to connect to Home Assistant" screen, sometimes I don't get the gear icon for access to settings,.  I have to do a three finger server switch gesture to get to settings.

**Comment by bgoncal** [comment]
There are 2 connections, one is the webview the other is the websocket which keeps things updated in the app for widgets, CarPlay, watch etc.

Websocket will try to connect in all servers you have, webview will be triggered as soon as you switch to that server.

Can you reach out to me in discord? (Bgoncal2)

I would like your internal and external URL configuration (screenshot) per server + to update to the latest beta

**Comment by rct** [comment]
Ok, PM'd 3 server configs on discord. (2 servers with cloud subscriptions and one internal test server)

**Comment by rct** [comment]
latest testing results:

- Updated to 2024.963. Opened app and it connected successfully. Was able to switch between y  3 servers no issues.
- Switched to other (safari, mail) apps and came back to Hass was no longer able to connect to any of the servers
- Forced closed Hass. Rexopened was able to successfully connect again to my 3 servers.
- Switched to other apps and came back to Hass app, the problem reappeared.

also confirmed that when I initially get the connection failed/retrying screen, the Gear icon does NOT appear until I do a 3 finger server switch, so that seems like a related bug?

Thanks for your help

**Comment by bgoncal** [comment]
Can you send me a screen recording of the above? Also export your logs and send to me once again please.

**Comment by rct** [comment]
Ok, I will reinstall 2024.963 and pull the screen recording together with matching logs for you.  Looking at the logs I did capture before they were too old there was a lot of these two lines:

```
[TokenManager.swift:152] refreshToken() > refresh token got error: sessionTaskFailed(error: Error Domain=NSURLErrorDomain Code=-1022 "The resource could not be loaded because the App Transport Security policy requires the use of a secure connection."
```

```
[Environment.swift:74] init() > WebSocket: delegate failed to provide access token sessionTaskFailed(error: Error Domain=NSURLErrorDomain Code=-1022 "The resource could not be loaded because the App Transport Security policy requires the use of a secure connection."
```

**Comment by rct** [comment]
Current status: The beta 2024.11 - build 2024.978 seems to fix the problem.   No connection issues or `Disconnected: Received close message during auth phase` errors since installing it. 

Also updated the original issue, through testing learned that the problem started with 2024.10 build 2024.927. The first two 2024.10 builds 2024.911 and 2024.913 did NOT have the problem.

**Comment by bgoncal** [comment]
Thanks for confirming the fix, I just merged the PR for it and it should be released within 2024.12 app version, the version on the App Store now doesn't contain the bug, so there is no hotfix needed.

**Comment by rct** [comment]
This may be a separate issue but while traveling today had some issues with 2024.978 connecting properly to the external (Nabu Casa) URLs with my iPad.

After a couple of manual retries it re-connected to the server it had been connected to previously (WHB). When I tried to switch to the other server (NYC) with a Nabu Casa sub, I couldn't get it to reconnect externally. I went on VPN it connected and I was able to verify the server (NYC) was working. Disconnected from VPN but still couldn't get it to connect to the external URL for the NYC server.  A manual retry (or two?) was needed to get it to connect back to the WHB server via Nabu Casa.

Also I will open a separate issue, I have the sensor widget showing 2 sensors which works fine while connected to the internal URLs.  When connected externally it shows the wrong sensor for one, and the wrong unit (deg C instead of deg F) for the other.
